src/proposer/proposer.cpp

```diff
@@ -29,15 +30,75 @@
 
 namespace proposer {
 
-class ProposerStub : public Proposer {
+static std::shared_ptr<const CBlock> GenerateBlock(staking::ActiveChain &active_chain,
```

Comment:
I believe the `static` keyword here effects that the symbol `proposer::GenerateBlock` is not exported by this `*.o` file. The C++ way of doing this, also according to our code style I believe, is by using an anonymous namespace.